sdk/pinocchio/src/memory.rs

```diff
@@ -47,33 +47,36 @@ pub unsafe fn sol_memcpy(dst: &mut [u8], src: &[u8], n: usize) {
 /// are of same type and `impl Copy`.
 ///
 /// This helper will be useful to optimize CU if
-/// copied size is > 32 bytes.
+/// copied size is `> 32` bytes.
 ///
-/// For value smaller than 32 bytes, `dst = src`
+/// For value smaller than `32` bytes, `dst = src`
 /// will be emitted to register assignment. For
-/// values larger than 32 bytes, compiler will
+/// values larger than `32` bytes, compiler will
 /// generate excess boilerplate to `sol_memcpy_`.
-/// So if T's size is known to be > 32 bytes,
+/// So if `T`'s size is known to be `> 32` bytes,
 /// this helper should be used.
 ///
 /// # Arguments
 ///
 /// - `dst` - Destination reference to copy to
 /// - `src` - Source reference to copy from
 #[inline]
-pub fn copy_val<T: ?Sized>(dst: &mut T, src: &T) {
+pub fn copy_val<T: Copy>(dst: &mut T, src: &T) {
```

Comment:
I believe this also needs to drop `dst` before copying over, otherwise `dst` may not be properly cleaned up